homeassistant/components/device_tracker/config_entry.py

```diff
@@ -164,6 +263,83 @@ def is_connected(self) -> bool:
         """Return true if the device is connected to the network."""
         raise NotImplementedError
 
+    @property
+    def unique_id(self) -> str | None:
+        """Return unique ID of the entity."""
+        return self.mac_address
+
+    @final
+    @property
+    def device_info(self) -> DeviceInfo | None:
+        """Device tracker entities should not create device registry entries."""
+        return None
+
+    @property
+    def entity_registry_enabled_default(self) -> bool:
+        """Return if entity is enabled by default."""
+        return self.find_device_entry() is not None
+
+    @callback
+    def add_to_platform_start(
+        self,
+        hass: HomeAssistant,
+        platform: EntityPlatform,
+        parallel_updates: asyncio.Semaphore | None,
+    ) -> None:
+        """Start adding an entity to a platform."""
+        super().add_to_platform_start(hass, platform, parallel_updates)
+        if self.mac_address:
+            _async_register_mac(
+                hass, platform.platform_name, self.mac_address, self.unique_id
+            )
+
+    @callback
+    def find_device_entry(self) -> dr.DeviceEntry | None:
+        """Return device entry."""
+        if self.mac_address is None:
+            return None
+
+        return dr.async_get(self.hass).async_get_device(
+            set(), {(dr.CONNECTION_NETWORK_MAC, self.mac_address)}
+        )
+
+    async def async_internal_added_to_hass(self) -> None:
+        """Handle added to Home Assistant."""
+        # Entities without a unique ID don't have a device
+        if (
+            not self.registry_entry
+            or not self.platform
+            or not self.platform.config_entry
+            or not self.mac_address
+            # Entities should not have a device info. We opt them out
+            # of this logic if they do.
+            or self.device_info
+        ):
+            if self.device_info:
+                LOGGER.warning("Entity %s unexpectedly has a device info")
+            await super().async_internal_added_to_hass()
+            return
+
+        device_entry = self.find_device_entry()
+        assert device_entry is not None
+
+        if device_entry is not None:
```

Comment:
The assert is not needed here.

```suggestion
        if (device_entry := self.find_device_entry()) is not None:
```